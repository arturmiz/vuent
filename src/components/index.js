import './Vnt.vue';
export { default as VntAutosuggest } from './autosuggest';
export { default as VntButton } from './button';
export { default as VntCheckbox } from './checkbox';
export { default as VntDialog } from './dialog';
export { default as VntHeader } from './header';
export { default as VntIcon } from './icon';
export { default as VntInput } from './input';
export { default as VntLink } from './link';
export { default as VntListView } from './listview';
export { default as VntNavView } from './navview';
export { default as VntPivot } from './pivot';
export { default as VntRadio } from './radio';
export { default as VntRating } from './rating';
export { default as VntSelect } from './select';
export { default as VntSlider } from './slider';
export { default as VntToggle } from './toggle';
