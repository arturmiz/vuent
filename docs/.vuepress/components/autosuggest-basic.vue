<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestBasic',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
