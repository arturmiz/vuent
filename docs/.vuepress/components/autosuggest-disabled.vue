<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries" disabled></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestDisabled',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
