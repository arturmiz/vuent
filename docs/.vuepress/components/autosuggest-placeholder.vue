<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"
                     placeholder="Type something..."></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestPlaceholder',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
