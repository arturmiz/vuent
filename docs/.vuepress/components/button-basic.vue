<template>
  <div class="playground">
    <vnt-button>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonBasic',
  components: {
    VntButton
  }
}
</script>
