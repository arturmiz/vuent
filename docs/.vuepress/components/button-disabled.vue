<template>
  <div class="playground">
    <vnt-button disabled>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonDisabled',
  components: {
    VntButton
  }
}
</script>
