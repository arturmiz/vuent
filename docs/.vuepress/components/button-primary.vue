<template>
  <div class="playground">
    <vnt-button primary>Primary</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonPrimary',
  components: {
    VntButton
  }
}
</script>
