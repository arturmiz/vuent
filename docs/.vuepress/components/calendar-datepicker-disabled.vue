<template>
  <div class="playground">
    <vnt-calendar-datepicker v-model="date" disabled />
    <div class="result">date: {{ date }}</div>
  </div>
</template>

<script>
import { VntCalendarDatepicker } from "../../../src/components";

export default {
  name: "CalendarDatepickerBasic",
  components: {
    VntCalendarDatepicker
  },
  data() {
    return {
      date: "2020-04-05"
    };
  }
};
</script>
