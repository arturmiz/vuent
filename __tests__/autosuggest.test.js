import { createLocalVue, mount } from '@vue/test-utils';
import { VntAutosuggest } from '@/components';
import { isInstalled } from './utils';

describe('Auto-suggest', () => {
  let wrapper;

  test('can be installed separately', () => {
    const localVue = createLocalVue();
    localVue.use(VntAutosuggest);

    expect(isInstalled(localVue, VntAutosuggest)).toBe(true);
  });

  describe('by default', () => {

    beforeAll(() => {
      wrapper = mount(VntAutosuggest);
    });

    test('is not disabled', () => {
      expect(wrapper.vm.disabled).toBe(false);
    });

    test('label text is empty', () => {
      expect(wrapper.vm.label).toBe('');
    });

    test('placeholder text is empty', () => {
      expect(wrapper.vm.placeholder).toBe('');
    });

    test('options array is empty', () => {
      expect(wrapper.vm.options).toHaveLength(0);
    });

    test('hints array is empty', () => {
      expect(wrapper.vm.hints).toHaveLength(0);
    });

    test('hints are hidden', () => {
      expect(wrapper.vm.showHints).toBe(false);
    });

    test('value is empty', () => {
      expect(wrapper.vm.value).toBe('');
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('can have custom label', () => {

    beforeAll(() => {
      wrapper = mount(VntAutosuggest, {
        propsData: {
          label: 'Custom Auto-suggest'
        }
      });
    });

    test('is correct', () => {
      expect(wrapper.vm.label).toBe('Custom Auto-suggest');
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('can have a custom placeholder', () => {

    beforeAll(() => {
      wrapper = mount(VntAutosuggest, {
        propsData: {
          placeholder: 'Type something'
        }
      });
    });

    test('is placeholder correct', () => {
      expect(wrapper.vm.placeholder).toBe('Type something');
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('can be disabled', () => {

    beforeAll(() => {
      wrapper = mount(VntAutosuggest, {
        propsData: {
          disabled: true,
          options: [
            'List item 1',
            'List item 2',
          ]
        }
      });
    });

    test('is disabled', () => {
      expect(wrapper.vm.disabled).toBe(true);
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    describe('when value has changed', () => {

      beforeAll(() => {
        const control = wrapper.find('.vnt-autosuggest__input');
        control.trigger('input', { value: 'lis' });
      });

      test('hints list is not shown', () => {
        expect(wrapper.vm.showHints).toBe(false);
      });

      test('hints list is not visible', () => {
        const list = wrapper.find('.vnt-dropdown-options');
        expect(list.isVisible()).toBe(false);
      });

      test('renders correctly', () => {
        expect(wrapper.html()).toMatchSnapshot();
      });

    });

  });

  describe('can have custom label and be disabled', () => {

    beforeAll(() => {
      wrapper = mount(VntAutosuggest, {
        propsData: {
          disabled: true,
          label: 'Custom Auto-suggest'
        }
      });
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('when have a value on init', () => {

    beforeAll(() => {
      wrapper = mount(VntAutosuggest, {
        propsData: {
          value: 'Li',
          options: [
            'List item 1',
            'List item 2',
          ]
        }
      });
    });

    test('hints list is not shown', () => {
      expect(wrapper.vm.showHints).toBe(false);
    });

    test('hints list is empty', () => {
      expect(wrapper.vm.hints).toHaveLength(0);
    });

    test('hints list is not visible', () => {
      const list = wrapper.find('.vnt-dropdown-options');
      expect(list.isVisible()).toBe(false);
    });

  });

  describe('when user types something', () => {

    describe('that match any option', () => {

      beforeAll(() => {
        wrapper = mount(VntAutosuggest, {
          propsData: {
            value: '',
            options: [
              'Austria',
              'Canada',
              'Australia',
              'Germany',
              'France',
              'United States',
            ]
          }
        });

        const input = wrapper.find('.vnt-autosuggest__input');
        input.trigger('input', { value: 'aus' });
      });

      test('returns hints', () => {
        const expectedHints = [
          'Australia',
          'Austria'
        ];
        expect(wrapper.vm.hints).toEqual(expectedHints);
      });

      test('event is emitted', () => {
        const [[eventValue]] = wrapper.emitted()['input'];
        expect(eventValue).toBe('aus');
      });

      test('hints list is shown', () => {
        expect(wrapper.vm.showHints).toBe(true);
      });

      test('hints list is visible', () => {
        const list = wrapper.find('.vnt-dropdown-options');
        expect(list.isVisible()).toBe(true);
      });

      test('renders correctly', () => {
        expect(wrapper.html()).toMatchSnapshot();
      });

    });

    describe('that doesn\'t match any option', () => {

      beforeAll(() => {
        wrapper = mount(VntAutosuggest, {
          propsData: {
            value: '',
            options: [
              'Austria',
              'Canada',
              'Australia',
              'Germany',
              'France',
              'United States',
            ]
          }
        });

        const input = wrapper.find('.vnt-autosuggest__input');
        input.trigger('input', { value: 'pol' });
      });

      test('returns no hints', () => {
        expect(wrapper.vm.hints).toHaveLength(0);
      });

      test('event is emitted', () => {
        const [[eventValue]] = wrapper.emitted()['input'];
        expect(eventValue).toBe('pol');
      });

      test('hints list is not shown', () => {
        expect(wrapper.vm.showHints).toBe(false);
      });

      test('hints list is not visible', () => {
        const list = wrapper.find('.vnt-dropdown-options');
        expect(list.isVisible()).toBe(false);
      });

      test('renders correctly', () => {
        expect(wrapper.html()).toMatchSnapshot();
      });

    });

  });

  describe('when selects a hint', () => {

    describe('and not disabled', () => {

      beforeAll(() => {
        wrapper = mount(VntAutosuggest, {
          propsData: {
            disabled: false
          }
        });

        wrapper.vm.selectHint('Austria');
      });

      test('event is emitted', () => {
        const [[eventValue]] = wrapper.emitted()['input'];
        expect(eventValue).toBe('Austria');
      });

    });

    describe('and is disabled', () => {

      beforeAll(() => {
        wrapper = mount(VntAutosuggest, {
          propsData: {
            disabled: true
          }
        });

        wrapper.vm.selectHint('Austria');
      });

      test('event is not emitted', () => {
        const inputEvent = wrapper.emitted()['input'];
        expect(inputEvent).toBeUndefined();
      });

    });

  });

});
