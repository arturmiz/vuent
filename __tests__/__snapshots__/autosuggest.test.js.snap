// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auto-suggest by default renders correctly 1`] = `
<div class="vnt-autosuggest">
  <!---->
  <div role="combobox" class="vnt-autosuggest__control">
    <input type="text" placeholder="" class="vnt-autosuggest__input">
    <ul role="listbox" class="vnt-dropdown-options" style="display: none;"></ul>
  </div>
</div>
`;

exports[`Auto-suggest can be disabled renders correctly 1`] = `
<div class="vnt-autosuggest">
  <!---->
  <div role="combobox" class="vnt-autosuggest__control">
    <input type="text" disabled="disabled" placeholder="" class="vnt-autosuggest__input">
    <ul role="listbox" class="vnt-dropdown-options" style="display: none;"></ul>
  </div>
</div>
`;

exports[`Auto-suggest can be disabled when value has changed renders correctly 1`] = `
<div class="vnt-autosuggest">
  <!---->
  <div role="combobox" class="vnt-autosuggest__control">
    <input type="text" disabled="disabled" placeholder="" class="vnt-autosuggest__input">
    <ul role="listbox" class="vnt-dropdown-options" style="display: none;"></ul>
  </div>
</div>
`;

exports[`Auto-suggest can have a custom placeholder renders correctly 1`] = `
<div class="vnt-autosuggest">
  <!---->
  <div role="combobox" class="vnt-autosuggest__control">
    <input type="text" placeholder="Type something" class="vnt-autosuggest__input">
    <ul role="listbox" class="vnt-dropdown-options" style="display: none;"></ul>
  </div>
</div>
`;

exports[`Auto-suggest can have custom label and be disabled renders correctly 1`] = `
<div class="vnt-autosuggest">
  <label class="vnt-autosuggest__label vnt-autosuggest__label--disabled">
    Custom Auto-suggest
  </label>
  <div role="combobox" class="vnt-autosuggest__control">
    <input type="text" disabled="disabled" placeholder="" class="vnt-autosuggest__input">
    <ul role="listbox" class="vnt-dropdown-options" style="display: none;"></ul>
  </div>
</div>
`;

exports[`Auto-suggest can have custom label renders correctly 1`] = `
<div class="vnt-autosuggest">
  <label class="vnt-autosuggest__label">
    Custom Auto-suggest
  </label>
  <div role="combobox" class="vnt-autosuggest__control">
    <input type="text" placeholder="" class="vnt-autosuggest__input">
    <ul role="listbox" class="vnt-dropdown-options" style="display: none;"></ul>
  </div>
</div>
`;

exports[`Auto-suggest when user types something that doesn't match any option renders correctly 1`] = `
<div class="vnt-autosuggest">
  <!---->
  <div role="combobox" class="vnt-autosuggest__control">
    <input type="text" placeholder="" class="vnt-autosuggest__input">
    <ul role="listbox" class="vnt-dropdown-options" style="display: none;"></ul>
  </div>
</div>
`;

exports[`Auto-suggest when user types something that match any option renders correctly 1`] = `
<div class="vnt-autosuggest">
  <!---->
  <div role="combobox" class="vnt-autosuggest__control">
    <input type="text" placeholder="" class="vnt-autosuggest__input">
    <ul role="listbox" class="vnt-dropdown-options" style="">
      <li role="listitem" class="vnt-dropdown-options__item">Australia</li>
      <li role="listitem" class="vnt-dropdown-options__item">Austria</li>
    </ul>
  </div>
</div>
`;
